File: 08/01/pages/LoginPage.js
'use strict';
var sleep = require('sleep');

var driver;
var USERNAME_INPUT = {id: 'username'};
var PASSWORD_INPUT = {id: 'password'};
var SUBMIT_BUTTON = {css: 'button'};
var SUCCESS_MESSAGE = {css: '.flash.success'};

function LoginPage(driver) {
  this.driver = driver;
  this.driver.get('http://the-internet.herokuapp.com/login');
}

LoginPage.prototype.with = function(username, password) {
  this.driver.findElement(USERNAME_INPUT).sendKeys(username);
  this.driver.findElement(PASSWORD_INPUT).sendKeys(password);
  this.driver.findElement(SUBMIT_BUTTON).click().then(function() {
    sleep.sleep(1);
  });
};

LoginPage.prototype.successMessagePresent = function() {
  return this.driver.findElement(SUCCESS_MESSAGE).isDisplayed();
};

module.exports = LoginPage;

File: csharp/08/03/PageObjects/LoginPage.cs
using OpenQA.Selenium;

namespace PageObjects
{
    class LoginPage
    {
        IWebDriver Driver;
        By LoginForm = By.Id("login");
        By UsernameInput = By.Id("username");
        By PasswordInput = By.Id("password");
        By SubmitButton = By.CssSelector("button");
        By SuccessMessage = By.CssSelector(".flash.success");
        By FailureMessage = By.CssSelector(".flash.error");

        public LoginPage(IWebDriver driver)
        {
            Driver = driver;
            Driver.Navigate().GoToUrl("http://the-internet.herokuapp.com/login");
            if (!Driver.FindElement(LoginForm).Displayed)
            {
                throw new System.Exception("Page not ready");
            }
        }

        public void With(string username, string password)
        {
            Driver.FindElement(UsernameInput).SendKeys(username);
            Driver.FindElement(PasswordInput).SendKeys(password);
            Driver.FindElement(SubmitButton).Click();
        }

        public bool SuccessMessagePresent()
        {
            return Driver.FindElement(SuccessMessage).Displayed;
        }

        public bool FailureMessagePresent()
        {
            return Driver.FindElement(FailureMessage).Displayed;
        }
    }
}
